### DISCUSSION TITLE
Is it really necessary to obtain and analyze the LG IRs of all Bloch states in energy bands?

### DISCUSSION THREAD

**Discussion by hongyi-zhao** [goodluck1982/SpaceGroupIrep#30] [General]
In traditional DFT calculation and result analysis, the most concerned is the band properties near the Fermi surface/level. After the rise of topological material science, band topology began to receive attention. Nevertheless, as the `irvsp` and `vasp2trace` tools have done, it is sufficient to determine the topological properties of materials using the trace data of occupied Bloch states.

But further, `SpaceGroupIrep` aims to realize calculation and analysis the LG IRs of all Bloch states (wave functions) in energy bands. IMO, the job done in `SpaceGroupIrep` has more systematic and theoretical significance. But I still don't quite understand the significance of considering the unoccupied states far away from the Fermi surface/level in the actual material research.

Regards,
HZ

**Comment by goodluck1982** [comment]
The anwser of whether it is necessary depends on the concrete researches of the users. Both `vasp2trace` and `irvsp` are goal-oriented tools. `vasp2trace` is written for checking the topological properties of materials, so only occupied states are enough. While `irvsp` is designed to give the LG IRs of vasp bands. But `SpaceGroupIrep` is a space-group-representation-theory-related database and tool sets, and its applications are open. Determining LG IRs in only one function of `SpaceGroupIrep`. All the same, of which bands the LG IRs are interested is different for vairous research purposes. This should be determined by users, not by `SpaceGroupIrep`. For examples, for spectrum of phonon and photonic crystal, there is no concepts such as "Fermi energy" and "occupied bands", and in which every band may be the interested one.

Please refer to the following discussion in https://github.com/goodluck1982/SpaceGroupIrep/issues/5#issuecomment-1023134619
![image](https://user-images.githubusercontent.com/48718961/153544659-a4762c21-c9e2-4ce8-8fe8-9bba493bc81b.png)
